src/app/modules/main/profile_section/contacts/models/contact_list.nim

```diff
@@ -116,23 +115,38 @@ QtObject:
     }.toTable
 
   proc addContactToList*(self: ContactList, contact: ContactsDto) =
+    let index = self.getContactIndexByPubkey(contact.id)
+    if index > -1:
+      return
     self.beginInsertRows(newQModelIndex(), self.contacts.len, self.contacts.len)
     self.contacts.add(contact)
     self.endInsertRows()
     self.countChanged()
 
-  proc hasAddedContacts(self: ContactList): bool {.slot.} = 
+  proc removeContactFromList*(self: ContactList, pubkey: string) =
+    let index = self.getContactIndexByPubkey(pubkey)
+    if index == -1:
+      return
+    self.beginRemoveRows(newQModelIndex(), index, index)
+    self.contacts.delete(index)
+    self.endRemoveRows()
+    self.countChanged()
+
+  proc hasAddedContacts(self: ContactList): bool {.slot.} =
     for c in self.contacts:
       if(c.isContact()): return true
     return false
 
+  # There are much better ways of notifying qml about this change than sending this signal. 
+  # It also may has an impact to the app performances since it's handled on multiple places on the qml side.
+  # Would be good to get rid of it durign refactor phase.
   proc contactChanged*(self: ContactList, pubkey: string) {.signal.}
 
   proc updateContact*(self: ContactList, contact: ContactsDto) =
     var found = false
     let topLeft = self.createIndex(0, 0, nil)
     let bottomRight = self.createIndex(self.contacts.len, 0, nil)
-    for c in self.contacts:
+    for c in self.contacts.mitems:
```

Comment:
Has the data structure changed so that we now need to access `mitems` instead of just `contacts`?